src/utils/resolveXYScales.js

```diff
@@ -300,7 +272,36 @@ export default function resolveXYScales(ComposedComponent) {
     _resolveLabels(props) {
 
     }
+    _resolveSpacing(props, Component, scaleType, domain, scale){
+      const propsSpacing = props.spacing || {};
+
+      // short-circuit if all spacings provided
+      if(hasAllSpacing(propsSpacing)) return propsSpacing;     
 
+      let spacing = omitNullUndefined(propsSpacing);
+
+      if(_.isFunction(Component.getSpacing)) {
+        const componentSpacing = omitNullUndefined(Component.getSpacing({scaleType, domain, scale, ...props}));
+        spacing = _.assign(componentSpacing, spacing);
+        if(hasAllSpacing(spacing)) return spacing;
+      }
+
+      // if Component has children,
+      // recurse through descendants to resolve their spacings the same way,
+      // and combine them into a single spacing, if there are multiple
+      if(React.Children.count(props.children)) {
+        let childspacings = mapOverChildren(props.children, this._resolveSpacing, 'props', 'type', scaleType, domain, scale);
+
+        const childspacing = _.fromPairs(['top', 'bottom', 'left', 'right'].map(k => {
```

Comment:
might be nice to break this into a `combineSpacings` function since we have a similar `combineDomains`. It's even more general than that really, but I cant think of a good generalizeable name... `combineDirectionalObjectByMax` or something 😛